integration-tests/contract-permits.spec.ts

```diff
@@ -0,0 +1,164 @@
+import { CONFIGS } from "./config";
+import { MichelsonMap, MichelCodecPacker } from "@taquito/taquito";
+import { importKey } from '@taquito/signer';
+import { permit_admin_42} from "./data/permit_admin_42";
+import { permit_admin_42_expiry} from "./data/permit_admin_42_expiry";
+import { permit_admin_42_set} from "./data/permit_admin_42_set";
+
+CONFIGS().forEach(({ lib, rpc, setup }) => {
+  const Tezos = lib;
+
+  Tezos.setPackerProvider(new MichelCodecPacker());
+
+  const FAUCET_KEY = {
+    "mnemonic": [
+      "swear",
+      "involve",
+      "mixture",
+      "pyramid",
+      "typical",
+      "swift",
+      "arch",
+      "mention",
+      "decline",
+      "mind",
+      "federal",
+      "lamp",
+      "coffee",
+      "weather",
+      "quote"
+    ],
+    // file deepcode ignore HardcodedNonCryptoSecret/test: <please specify a reason of ignoring this>
+    "secret": "782ff5c78fc58df062670fdc27d5a8a8003ecf16",
+    "amount": "32252493855",
+    "pkh": "tz1Xk7HkSwHv6dTEgR7E2WC2yFj4cyyuj2Gh",
+    "password": "zXuYCjc8aq",
+    "email": "fnctuxjz.erhkqxca@tezos.example.org"
+       }
+
+    importKey(
+      Tezos,
+      FAUCET_KEY.email,
+      FAUCET_KEY.password,
+      FAUCET_KEY.mnemonic.join(' '),
+      FAUCET_KEY.secret
+    );
+
+  describe(`Test of contracts having a permit for tzip-17: ${rpc}`, () => {
+
+    beforeEach(async (done) => {
+      await setup()
+        done()
+    })
+
+  test('Permit can be submitted and used', async (done) => {
+
+      const op = await Tezos.contract.originate({
+        code: permit_admin_42,
+        storage:
+        {
+        0: new MichelsonMap(),
+        1: 0,
+        2: 'tz1bDCu64RmcpWahdn9bWrDMi6cu7mXZynHm'
+        },
+        });
+        await op.confirmation();
+        expect(op.hash).toBeDefined();
+        expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+        const contract = await op.contract();
+        expect(op.status).toEqual('applied')
+
+        const permit_parameter = 'Pair "edpkuPTVBFtbYd6gZWryXypSYYq6g7FvyucwphoU78T1vmGkbhj6qb" (Pair "edsigtfkWys7vyeQy1PnHcBuac1dgj2aJ8Jv3fvoDE5XRtxTMRgJBwVgMTzvhAzBQyjH48ux9KE8jRZBSk4Rv2bfphsfpKP3ggM" 0x0f0db0ce6f057a8835adb6a2c617fd8a136b8028fac90aab7b4766def688ea0c)'
+
+        //submit the permit_parameter to the contract
+        Tezos.contract
+        .at(contract.address)
+        .then((c) => {
+          return c.methods.permit(permit_parameter).send();
+        })
+        .then(async (op) => {
+            await op.confirmation(3);
+            return op.hash;
+          })
+          .catch(Error) ;
```

Comment:
The parameter of the permit method is incorrect, it should not be in Michelson. 
This call throws the following exception: `Invalid parameters error: permit Received 1 argument while expecting one of the following signatures ([["key","signature","bytes"]])` 
Is there a reason why the exception is caught?

I don't think that we should use `.catch(Error) ;` because this code is throwing an exception that is silently caught. I think we should make sure that the call is successful by testing if the hash of the operation is defined and that is status is `applied`.
It might be easier to reason about using the `await` syntax instead of `then`.
For example, I would replace line 74 to 83 by:
```
const permitMethodCall = await contract.methods.permit(permit_parameter).send();
await permitMethodCall.confirmation();

expect(permitMethodCall.hash).toBeDefined();
expect(permitMethodCall.status).toEqual('applied');
```

Same comment for the other places where  `.catch(Error) ;` is used.

Reply:
Thanks. I got the try-catch example from the Taquito help docs and I have been having trouble with it (getting asserts in). 
I have replaced the lines as suggested. Way better thanks!